clients/src/main/java/org/apache/kafka/common/KafkaFuture.java

```diff
@@ -59,10 +60,14 @@ public synchronized void accept(R newValue, Throwable exception) {
                 future.completeExceptionally(exception);
             } else {
                 remainingResponses--;
-                if (remainingResponses <= 0)
-                    future.complete(null);
+                maybeComplete();
             }
         }
+
+        private void maybeComplete() {
+            if (remainingResponses <= 0)
```

Comment:
It's just a generic robustness thing.  Why check for == 0 when you can just as easily check for <= 0?  I agree it should never happen.